src/vs/workbench/browser/parts/editor/editor.contribution.ts

```diff
@@ -132,15 +136,21 @@ class UntitledEditorInputFactory implements IEditorInputFactory {
 		return JSON.stringify(serialized);
 	}
 
-	public deserialize(instantiationService: IInstantiationService, serializedEditorInput: string): UntitledEditorInput {
-		return instantiationService.invokeFunction<UntitledEditorInput>(accessor => {
+	public deserialize(instantiationService: IInstantiationService, serializedEditorInput: string): EditorInput {
```

Comment:
Why do we need to change the type definition on this?

Reply:
If it's sql file, it needs to return as QueryResultsInput, otherwise return as UntitledEditorInput type. UntitledEditorInput doesn't understand connection

---
In reply to: [158188741](https://github.com/Microsoft/sqlopsstudio/pull/385#discussion_r158188741) [](ancestors = 158188741)